### PR TITLE
Lectures 06 and 07

### PR DESCRIPTION
**Pull request by trgardos**
Ok. Ready for review. I think L07 Clustering in Practice needs more improvements on the newsgroup example at the end, but it's a bit better now than it was.

Also, the figures are rendering quite large on the website, which I'm not sure how to fix yet. But we can fix that later.

### CODE REVIEW

**Comment by sladenheim** [comment]
@trgardos you can set figure properties using

{fig-align="center" width=80%}

I'll review this when I come in this morning.

**Comment by trgardos** [comment]
Thanks. Where do you put that? On a fenced div around the python code cell? Or right in the ```{python fig-align=”center” width=80%}?

From: sladenheim ***@***.***>
Date: Thursday, August 29, 2024 at 6:56 AM
To: tools4ds/DS701-Course-Notes ***@***.***>
Cc: Gardos, Thomas ***@***.***>, Mention ***@***.***>
Subject: Re: [tools4ds/DS701-Course-Notes] Lectures 06 and 07 (PR #47)

@trgardos<https://github.com/trgardos> you can set figure properties using

[X]<fig/fig.png>{fig-align="center" width=80%}

I'll review this when I come in this morning.

File: ds701_book/06-Clustering-I-kmeans.qmd [hunk 1]
```diff
@@ -312,19 +321,173 @@ The other from the top 10 were SVM, Apriori, EM, PageRank, AdaBoost, kNN, Naive
...
+    - In other words, $c_j$ is the mean of the vectors in $C_j$.
+4. Repeat (ie, go to Step 2) until convergence
+    - either the cluster centers change below a threshold, or inertia changes below a threshold or a maximum number of iterations is reached.
 
 ---
 
-Let's see this in practice:
+Let's see this in practice with well separated clusters and also look at the WCSS:
+
+```{python}
```
Comment (sladenheim) [inline ds701_book/06-Clustering-I-kmeans.qmd:338]: You can put these lines under ```{python}

```
#| fig-align: center
#| fig-width: 75%
#| fig-wdith: 75%
```

More details on figs here 

- https://quarto.org/docs/authoring/figures.html#computations
- https://quarto.org/docs/computations/execution-options.html#figure-options

Comment (trgardos) [reply ds701_book/06-Clustering-I-kmeans.qmd:338]: didn't work. can we merge and fix it later?

File: ds701_book/06-Clustering-I-kmeans.qmd [hunk 2]
```diff
@@ -573,3 +905,14 @@ Finally, we use 1 bit (2 color groups) for a compression ratio of $24\times$.
     
 ![](figs/L6-annie_002.png){width=500 fig-align="center"}
 
+## Recap and Next
+
+Today we covered:
+
+* $k$-means clustering
+* strengths and weaknesses
+* Importance of initialization and cluster numberr
```
Comment (sladenheim) [inline ds701_book/06-Clustering-I-kmeans.qmd:914]: typo with numberr

File: ds701_book/07-Clustering-II-in-practice.qmd [hunk 3]
```diff
@@ -3,88 +3,219 @@ title: Clustering In Practice
...
+
+* Metric MDS, of which Classical MDS is a special case, and has a closed form solution
+  based on the eigenvectors of the centered distance matrix
+* Non-Metric MDS, which tries to find a non-parametric monotonic relationship
+  between the dissimilarities and the target coordinates through an iterative approach
+:::
+
+::: {.fragment}
+Also note that MDS may not always work well if, for example the dissimilarities
+are not well modeled by a metric like euclidean distance
```
Comment (sladenheim) [inline ds701_book/07-Clustering-II-in-practice.qmd:212]: Should we capitalize Euclidean?

File: ds701_book/07-Clustering-II-in-practice.qmd [hunk 4]
```diff
@@ -141,167 +318,277 @@ Thus, the general sequence for using a tool from `scikit-learn` is:
...
 
+Here again, you can see the inter-cluster sample distances are small.
+
 ## Cluster Evaluation
 
 How do we know whether the clusters we get represent "real" structure in our data?
 
-Consider this dataset, which we have clustered using $k$-means:
+:::: {.fragment}
+Consider a dataset of 100 points _uniformly distirbuted_ in the unit square.
```
Comment (sladenheim) [inline ds701_book/07-Clustering-II-in-practice.qmd:362]: typo on distributed

File: ds701_book/07-Clustering-II-in-practice.qmd [hunk 5]
```diff
@@ -141,167 +318,277 @@ Thus, the general sequence for using a tool from `scikit-learn` is:
...
+## Rand Index Defined
 
 Specifically:
 
-Let $a$ be the number of pairs that have the same label in $T$ and the same label in $C$. 
-
-Let $b$ be: the number of pairs that have different labels in $T$ and different labels in $C$. 
+::: {.incremental}
+- Let $a$ be the number of pairs that have the same label in $T$ and the same label in $C$. 
+- Let $b$ be: the number of pairs that have different labels in $T$ and different labels in $C$.
```
Comment (sladenheim) [inline ds701_book/07-Clustering-II-in-practice.qmd:534]: Is the colon meant to be here?

File: ds701_book/07-Clustering-II-in-practice.qmd [hunk 6]
```diff
@@ -324,44 +611,48 @@ The second question we face in evaluating a clustering is how many clusters are
...
-#| tags: [hide-input]
 plt.plot(range(1, len(error)), error[1:], 'o-')
 plt.xlabel('Number of clusters')
 plt.title(r'$k$-means clustering performance of synthetic data')
 plt.ylabel('Error');
 ```
 
-__Warning__: This synthetic data is not at all typical.   You will almost never see such a sharp change in the error function as we see here.
+::: {.callout-warning}
+This synthetic data is not at all typical.   You will almost never see such a sharp change in the error function as we see here.
```
Comment (sladenheim) [inline ds701_book/07-Clustering-II-in-practice.qmd:647]: We can remove the space after the period here.

File: ds701_book/07-Clustering-II-in-practice.qmd [hunk 7]
```diff
@@ -426,175 +743,201 @@ plt.title('$k$-means clustering performance on synthetic data')
...
-The starting point is to note that one can encode a set using a binary vector with one component for each potential set member.  
+The so-called _bag of words_ encoding for a document is to treat the document as a multiset of words.
 
-The so-called _bag of words_ encoding for a document is to treat the document as a **multi**set of words.
+::: aside
+A _multiset_ is like a set but allows for multiple instances of each element.
+:::
 
-That is, we simply count how many times each word occurs.   It is a "bag" because all the order of the words in the document is lost.
+That is, we simply count how many times each word occurs.   It is a "bag" because all the order of the words in the
```
Comment (sladenheim) [inline ds701_book/07-Clustering-II-in-practice.qmd:810]: Can delete extra space.

File: ds701_book/07-Clustering-II-in-practice.qmd [hunk 8]
```diff
@@ -426,175 +743,201 @@ plt.title('$k$-means clustering performance on synthetic data')
...
+document is lost.
 
 Surprisingly, we can still tell a lot about the document even without knowing its word ordering.
 
+---
+
 Counting the number of times each word occurs in a document yields a vector of __term frequencies.__
 
-However, simply using the "bag of words" directly has a number of drawbacks.   First of all, large documents will have more words than small documents.   
+However, simply using the "bag of words" directly has a number of drawbacks.   First of all, large documents will have
```
Comment (sladenheim) [inline ds701_book/07-Clustering-II-in-practice.qmd:819]: Remove additional space.

**Review by sladenheim — APPROVED** [review]
Just a few small typos to fix up.

File: ds701_book/06-Clustering-I-kmeans.qmd (hunk 1 above)
Comment (sladenheim) [reply ds701_book/06-Clustering-I-kmeans.qmd:338]: Yes, let's leave it to fix later. Just finished my review.

File: ds701_book/07-Clustering-II-in-practice.qmd (hunk 8 above)
Comment (trgardos) [reply ds701_book/07-Clustering-II-in-practice.qmd:819]: done.

File: ds701_book/07-Clustering-II-in-practice.qmd (hunk 7 above)
Comment (trgardos) [reply ds701_book/07-Clustering-II-in-practice.qmd:810]: Done.

File: ds701_book/07-Clustering-II-in-practice.qmd (hunk 6 above)
Comment (trgardos) [reply ds701_book/07-Clustering-II-in-practice.qmd:647]: Done.

File: ds701_book/07-Clustering-II-in-practice.qmd (hunk 5 above)
Comment (trgardos) [reply ds701_book/07-Clustering-II-in-practice.qmd:534]: done

File: ds701_book/07-Clustering-II-in-practice.qmd (hunk 4 above)
Comment (trgardos) [reply ds701_book/07-Clustering-II-in-practice.qmd:362]: done.

File: ds701_book/07-Clustering-II-in-practice.qmd (hunk 3 above)
Comment (trgardos) [reply ds701_book/07-Clustering-II-in-practice.qmd:212]: done

File: ds701_book/06-Clustering-I-kmeans.qmd (hunk 2 above)
Comment (trgardos) [reply ds701_book/06-Clustering-I-kmeans.qmd:914]: done

File: ds701_book/06-Clustering-I-kmeans.qmd (hunk 1 above)
Comment (trgardos) [reply ds701_book/06-Clustering-I-kmeans.qmd:338]: ok

**Pull request merged**